tests/View/Blade/AntlersComponents/PartialCompilerTest.php

```diff
@@ -0,0 +1,382 @@
+
+        $template = <<<'TEMPLATE'
+<s:partial:the_partial :when="$theValue" />
+TEMPLATE;
+
+        $this->assertSame('', Blade::render($template, ['theValue' => false]));
+        $this->assertSame('The content', Blade::render($template, ['theValue' => true]));
+    }
+
+    #[Test]
+    public function it_compiles_unless_parameter()
+    {
+        $this->viewShouldReturnRaw('the_partial', 'The content');
+
+        $template = <<<'TEMPLATE'
+<s:partial:the_partial :unless="$theValue" />
+TEMPLATE;
+
+        $this->assertSame('', Blade::render($template, ['theValue' => true]));
+        $this->assertSame('The content', Blade::render($template, ['theValue' => false]));
+    }
+
+    #[Test]
+    public function it_compiles_conditional_parameters_with_slots()
+    {
+        $alert = <<<'ALERT'
+<div id="header">{{ $header }}</div>
+<div>{{ $slot }}</div>
+<div id="footer">{{ $footer }}</div>
+ALERT;
+        $this->viewShouldReturnRaw('alert', $alert);
+
+        $template = <<<'BLADE'
+<s:partial:alert :when="$theValue">
+  <s:slot:header>The header</s:slot:header>
+  <s:slot.footer>The footer</s:slot.footer>
+  I am the slot content.
+</s:partial:alert>
+BLADE;
+
+        $expected = <<<'EXPECTED'
+<div id="header">The header</div>
+<div>I am the slot content.</div>
+<div id="footer">The footer</div>
+EXPECTED;
+
+        $this->assertSame('', Blade::render($template, ['theValue' => false]));
+        $this->assertSame($expected, Blade::render($template, ['theValue' => true]));
+    }
+
+    #[Test]
+    public function it_compiles_nested_partials()
+    {
+        $alert = <<<'ANTLERS'
+<div id="header">{{ $header }}</div>
+<div>{{ slot }}</div>
+<div id="footer">{{ $footer }}</div>
+ANTLERS;
+
+        $this->viewShouldReturnRaw('alert', $alert);
+
+        $template = <<<'BLADE'
+<s:partial:alert>
+  <s:slot:header>The header</s:slot:header>
+  <s:slot.footer>The footer</s:slot.footer>
+  I am the slot content.
+  
+  <s:partial:alert>
+    <s:slot:header>The header</s:slot:header>
+    <s:slot.footer>The footer2</s:slot.footer>
+    I am the second slot content.
+  </s:partial:alert>
+</s:partial:alert>
+BLADE;
+
+        $expected = <<<'EXPECTED'
+<div id="header">The header</div>
+<div>I am the slot content.
+  
+  <div id="header">The header</div>
+<div>I am the second slot content.</div>
+<div id="footer">The footer2</div></div>
+<div id="footer">The footer</div>
+EXPECTED;
+
+        $this->assertSame(
+            $expected,
+            Blade::render($template)
+        );
+    }
+
+    #[Test]
+    public function it_merges_frontmatter_using_directive()
+    {
+        $this->withFakeViews();
+
+        $partial = <<<'BLADE'
+@frontmatter([
+  'name'  => 'The Name!',
+  'image' => 'https://example.com/placeholder.png',
+])
+
+{{ $name }} {{ $image }}
+BLADE;
+
+        $this->viewShouldReturnRaw('the_partial', $partial, 'blade.php');
+
+        $this->assertSame(
+            'The Name! https://example.com/placeholder.png',
```

Comment:
This should be empty.

Partial front matter is only available within the `view` array.
Partial params will be available as the passed-in name, but also override the view var.

e.g.
```
---
foo: in front matter
---
{{ foo }} / {{ view:foo }}
```
```
{{ partial:test }}                 " / in front matter"
{{ partial:test foo="in param" }}  "in param / in param"
```

Reply:
Good catch 🙌
Corrected w/ updated tests